lib/AST/DiagnosticBridge.cpp

```diff
@@ -77,6 +77,13 @@ static void addQueueDiagnostic(void *queuedDiagnostics,
                                    documentationPath.size(),
                                    highlightRanges.data(),
                                    highlightRanges.size() / 2);
+
+  // TODO: A better way to do this would be to pass the notes as an
+  // argument to `swift_ASTGen_addQueuedDiagnostic` but that requires
+  // bridging of `Note` structure and new serialization.
+  for (auto *childNote : info.ChildDiagnosticInfo) {
+    addQueueDiagnostic(queuedDiagnostics, *childNote, SM);
+  }
```

Comment:
Might be clearer to move this up into `enqueueDiagnostic`?

Reply:
I'd prefer to keep this in here for now since this is technically an implementation detail a "primary" diagnostic and ideally should be dispatched as an argument like what I mentioned in a TODO.

Reply:
Unless I'm misunderstanding, the infrastructure currently assumes that notes are attached to the first primary diagnostic in the group. Ie. `enqueueDiagnostic` isn't necessarily a primary diagnostic anyway. But 🤷‍♂️

Reply:
You might be talking about lower level things but there is a dedicated API in DiagnosticEngine which let's the users attach notes directly to the diagnostic they want. I think we should replace all of the disjoint emission of notes with `diagnoseWithNotes` and remove any inference that happens later on...

Reply:
Yeah, all for moving to `diagnoseWithNotes`. Happy to leave it where it is 👍